File: app/src/androidTest/java/com/star_zero/example/androidmvvm/MockApp.java
package com.star_zero.example.androidmvvm;

import com.star_zero.example.androidmvvm.di.AppComponent;
import com.star_zero.example.androidmvvm.di.DaggerMockAppComponent;
import com.star_zero.example.androidmvvm.di.MockAppModule;

public class MockApp extends App {

    @Override
    public AppComponent getAppComponent() {
        return DaggerMockAppComponent.builder()
                .mockAppModule(new MockAppModule())
                .build();
    }
}

File: app/src/androidTest/kotlin/com/star_zero/example/androidmvvm/MockApp.kt
package com.star_zero.example.androidmvvm

import com.star_zero.example.androidmvvm.di.DaggerMockAppComponent

class MockApp : App() {

    override fun onCreate() {
        super.onCreate()

        DaggerMockAppComponent.builder().build().inject(this)
    }
}
